spring-commercetools/src/main/java/com/epam/esm/commercetools/UpdateActionsBuilder.java

```diff
@@ -0,0 +1,124 @@
+
+        return Attribute
+                .builder()
+                .name(TAGS)
+                .value(tagsNames)
+                .build();
+    }
+
+    public List<ProductUpdateAction> preparingUpdatesForGiftCertificate(GiftCertificate giftCertificate, String id) {
+        Map<Predicate<GiftCertificate>, Consumer<List<ProductUpdateAction>>> updatesMap = new HashMap<>();
+
+        List<ProductUpdateAction> productUpdateActions = new ArrayList<>();
+
+        updatesMap.put(cert -> giftCertificate.getName() != null,
+                list -> list.add(ProductUpdateAction
+                        .changeNameBuilder()
+                        .name(LocalizedString
+                                .ofEnglish(giftCertificate.getName()))
+                        .build()));
+
+        updatesMap.put(cert -> giftCertificate.getDescription() != null,
+                list -> list.add(ProductUpdateAction
+                        .setDescriptionBuilder()
+                        .description(LocalizedString
+                                .ofEnglish(giftCertificate.getDescription()))
+                        .build()));
+
+        updatesMap.put(cert -> giftCertificate.getPrice() != null,
+                list -> list.add(ProductUpdateAction
+                        .setPricesBuilder()
+                        .variantId(1L)
+                        .prices(PriceDraft
+                                .builder()
+                                .value(Money
+                                        .builder()
+                                        .currencyCode(CURRENCY_CODE)
+                                        .centAmount(Long.parseLong(String.valueOf(giftCertificate.getPrice())))
+                                        .build())
+                                .build())
+                        .build()));
+
+        updatesMap.put(cert -> giftCertificate.getDurationDays() != null,
+                list -> {
+                    Attribute durationAttribute = setDurationAttribute(giftCertificate.getDurationDays());
+
+                    list.add(ProductUpdateAction
+                            .setAttributeBuilder()
+                            .variantId(1L)
+                            .name(durationAttribute.getName())
+                            .value(durationAttribute.getValue())
+                            .build());
+                });
+        updatesMap.put(cert -> giftCertificate.getTags() != null,
+                list -> {
+                    Attribute tagAttribute = setTagsAttributeWhenUpdateGiftCertificate(giftCertificate.getTags(), id);
+
+                    list.add(ProductUpdateAction
+                            .setAttributeBuilder()
+                            .variantId(1L)
+                            .name(tagAttribute.getName())
+                            .value(tagAttribute.getValue())
+                            .build());
+                });
+
+        updatesMap.forEach((key, value) -> {
```

Comment:
Define `updatesMap` outside of this method, to not create it each time it called

Reply:
done